theano/tensor/tests/test_sort.py

```diff
@@ -219,3 +235,247 @@ def test_argsort_grad():
+        x = tensor.vector(name='x', dtype=dtype)
+        fn = theano.function([x], argtopk(x, 1, axis=axis, idx_dtype=idx_dtype))
+        xval = np.asarray([1]).astype(dtype)
+        yval = fn(xval)
+        assert yval == np.asarray([0], dtype=idx_dtype)
+        assert yval.dtype == np.dtype(idx_dtype)
+
+    @utt.parameterized.expand(product(
+        _dtypes, [-1, 0, None]))
+    def test_topk_sanity(self, dtype, axis):
+        x = tensor.vector(name='x', dtype=dtype)
+        fn = theano.function([x], topk(x, 1, axis=axis))
+        xval = np.asarray([1]).astype(dtype)
+        yval = fn(xval)
+        assert yval == xval
+        assert yval.dtype == xval.dtype
+
+    @utt.parameterized.expand(product(
+        _dtypes, _int_dtypes, [-1, 0, None]))
+    def test_combined_sanity(self, dtype, idx_dtype, axis):
+        x = tensor.vector(name='x', dtype=dtype)
+        yv, yi = topk_and_argtopk(x, 1, axis=axis, idx_dtype=idx_dtype)
+        fn = theano.function([x], [yv, yi])
+        xval = np.asarray([1]).astype(dtype)
+        yvval, yival = fn(xval)
+        assert yival == np.asarray([0], dtype=idx_dtype)
+        assert np.allclose(xval, yvval)
+        assert yvval.dtype == xval.dtype
+        assert yival.dtype == np.dtype(idx_dtype)
+
+    @utt.parameterized.expand(chain(
+        product(
+            (16, 61, 257),
+            (1, -1, 10, -10, 'n//2', 'n-1', '-n', '1-n'),
+            ('float64', 'float16', 'int16', 'int8')),
+        ((2049, 1337, 'float64'),)))
+    def test_topk_1d(self, size, k, dtype):
+        if isinstance(k, str):
+            k = eval(k.replace('n', str(size)))
+
+        x = theano.tensor.vector(name='x', dtype=dtype)
+        y = topk(x, k)
+        fn = theano.function([x], y)
+        # generate a all-unique array
+        xval = gen_unique_vector(size, dtype)
+        yval = fn(xval)
+        idx = slice(-k, None) if k > 0 else slice(-k)
+        goal = np.sort(xval)[idx]
+
+        print(np.sort(yval))
+        print(goal)
+        assert yval.dtype == goal.dtype
+        assert np.allclose(np.sort(yval), goal)
+
+    @utt.parameterized.expand(chain(
+        product(
+            (16, 61, 257),
+            (1, -1, 10, -10, 'n//2', 'n-1', '-n', '1-n'),
+            ('float32', 'int32'),
+            ('int32', 'int64')),
+        ((2049, 1337, 'float32', 'int32'),)))
+    def test_argtopk_1d(self, size, k, dtype, idx_dtype):
+        if isinstance(k, str):
+            k = eval(k.replace('n', str(size)))
+
+        x = theano.tensor.vector(name='x', dtype=dtype)
+        y = argtopk(x, k, idx_dtype=idx_dtype)
+        fn = theano.function([x], y)
+        # generate a all-unique array
+        xval = gen_unique_vector(size, dtype)
+        yval = fn(xval)
+        idx = slice(-k, None) if k > 0 else slice(-k)
+        goal = np.argsort(xval)[idx].astype(idx_dtype)
+
+        # due to uniqueness, we expect indices same
+        assert np.all(xval[np.sort(yval)] == xval[np.sort(goal)])
```

Comment:
numpy garantie that if 2 values are the same, we will give the first indices. Do you keep that feature? Also on GPU? It should be clear in the doc.